sql/internal/sqlx/sqlx.go

```diff
@@ -277,7 +288,12 @@ func (b *Builder) mayQualify(s *schema.Schema, ident string) *Builder {
 		b.Ident(s.Name)
 		b.rewriteLastByte('.')
 	}
-	b.Ident(ident)
+	for i, ident := range idents {
```

Comment:
idents cannot be an empty slice, because we rely on it to have at least one (we call `rewriteLastByte` above). 

What do you say about changing its signature to: 

```go
mayQualify(s *schema.Schema, top string, children ...string) *Builder
```